web/src/buddy_list.ts

```diff
@@ -354,7 +354,53 @@ export class BuddyList extends BuddyListConf {
         }
     }
 
+    update_section_header_counts(): void {
+        const {
+            total_human_subscribers_count,
+            other_users_count,
+            hide_headers,
+            participant_ids_set,
+        } = this.render_data;
+        const subscribers_count = total_human_subscribers_count - this.participant_user_ids.length;
+
+        // We only show the participants list if it has members, so updating a count
+        // can affect if we show/hide this section.
+        const show_participants_list = !hide_headers && participant_ids_set.size;
+        $("#buddy-list-participants-container").toggleClass("no-display", !show_participants_list);
```

Comment:
Can you share more of this code with the caller? These first few lines at least look the same, and are complex enough to deserve their own comments.

Reply:
I might be misunderstanding something, but these lines all feel distinct to this function.

(1) Getting values from `this.render_data` seemed better than passing the values to this function, and I wouldn't think of it as complex.

(2) `total_human_subscribers_count - this.participant_user_ids.length` appears in a call to `render_section_header` below but didn't feel worth sharing with the caller. I should maybe rename `const subscribers_count` with `subscriber_section_user_count` or something to be clearer about the calculation.

(3) That third section has a comment, so I'm not sure if you were wanting more comment there or just to move it into the other function. I can see how we can do the `show_participants_list` either way, so I'll move that to share with the caller.

Reply:
`show_participants_list` was what I was certain about; wasn't sure if any of the other logic could be managed better as well.